lib/fragment.js

```diff
@@ -80,33 +83,53 @@ module.exports = class Fragment extends EventEmitter {
      * Handles fetching the fragment
      * @param {object} request - HTTP request stream
      * @param {boolean} isFallback - decides between fragment and fallback URL
-     * @returns {object} Fragment response streams in case of synchronous fragment or buffer incase of async fragment
+     * @param {object} parentSpan - opentracing Span that will be the parent of the current operation
+     * @returns {object} Fragment response streams in case of synchronous fragment or buffer in case of async fragment
      */
-    fetch(request, isFallback) {
+    fetch(request, isFallback, parentSpan = null) {
         if (!isFallback) {
             this.emit('start');
         }
         const url = isFallback
             ? this.attributes.fallbackUrl
             : this.attributes.url;
 
-        this.requestFragment(url, this.attributes, request).then(
-            res => this.onResponse(res, isFallback),
-            err => {
-                if (!isFallback) {
-                    const { fallbackUrl } = this.attributes;
-                    if (fallbackUrl) {
-                        this.emit('fallback', err);
-                        this.fetch(request, true);
+        const spanOptions = parentSpan ? { childOf: parentSpan } : {};
+        const span = tracer.startSpan('fetch_fragment', spanOptions);
+        span.setTag('fragment-url', url);
```

Comment:
can we use addTags instead of setTag here?

Reply:
I don't think the opentracing API spec has a addTags. This is also a deliberate attempt at using the spec as is to also understand how much such approach pollutes the codebase. We could export already a helper function in the tracing.js for that purpose. 
Something like:
```javascript
addTags(span, {...});
```
Would that work for you?